db/view/view.cc

```diff
@@ -786,14 +877,29 @@ void view_updates::update_entry(const partition_key& base_key, const clustering_
         return;
     }
 
-    deletable_row& r = get_view_row(base_key, update);
-    auto marker = compute_row_marker(update);
-    r.apply(marker);
-    r.apply(update.tomb());
+    auto view_rows = get_view_rows(base_key, update, std::nullopt);
+    auto update_marker = compute_row_marker(update);
+    for (const auto& [r, action] : view_rows) {
+        if (!action.apply_row_marker(r, now)) {
+            r->apply(update_marker);
+        }
+        r->apply(update.tomb());
+
+        auto diff = update.cells().difference(*_base, column_kind::regular_column, existing.cells());
+        add_cells_to_view(*_base, *_view, std::move(diff), r->cells());
+    }
+    _op_count += view_rows.size();
+}
 
-    auto diff = update.cells().difference(*_base, column_kind::regular_column, existing.cells());
-    add_cells_to_view(*_base, *_view, std::move(diff), r.cells());
-    _op_count++;
+void view_updates::update_entry_for_computed_column(
+        const partition_key& base_key,
+        const clustering_row& update,
+        const std::optional<clustering_row>& existing,
+        gc_clock::time_point now) {
+    auto view_rows = get_view_rows(base_key, update, existing);
+    for (const auto& [r, action] : view_rows) {
+        action.apply_all(r, now);
```

Comment:
For instance, her, instead of `apply_all`, we could just visit `action._action` and act accordingly. In my humble opinion, it would be way clearer for the reviewer than the current way with `apply_visitor`.

Reply:
apply_all is used so that we can provide transparently `now` for the shadowable tombstone. If you have any better ideas, please recommend.

Reply:
I don't follow your response - is it not trivial to std::visit the action here and open-code whatever's necessary? It adds a few more lines, but would also be way more self-descriptive - e.g. from looking at the snippet above, it's unclear what "all" means, and that it actually visits the action variant, applying whatever's needed in the action.